src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -364,14 +369,18 @@ public function getCompiler()
      */
     public function set($id, $service)
     {
-        $id = strtolower($id);
+        $caseSensitiveId = (string) $id;
```

Comment:
If that's the reason, I would revert this as `$id` is documented as being a string.